activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -51,12 +51,20 @@ def dealloc(stmt)
           end
       end
 
+      class Version < AbstractAdapter::Version # :nodoc:
```

Comment:
Do we need this special class?

Reply:
I added it because `full_version_string` defaults to `nil` in the abstract class.

https://github.com/rails/rails/blob/6584fb3939dd3892834ed93fa791064d5299cda2/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L175

Defaulting to `nil` is OK for other adapters like `SQLite3Adapter`, but `full_version_string` should always be set for adapters that inherit from `AbstractMysqlAdapter`.

I added `AbstractMysqlAdapter::Version` to make `full_version_string` required. I was on the fence about it, but I thought it'd be helpful. I'd be fine with removing it too.

Reply:
It is `AbstractMysqlAdapter` that is creating the version and it is always passing the value, so I don't think we need to make it required.